.github/CONTRIBUTING.md

```diff
@@ -0,0 +1,81 @@
+# Contributing to Tangle UI
+
+Thank you for your interest in contributing! We appreciate your time and effort. All contributions, from small typo fixes to large features, are valuable.
+
+## Local setup
+
+See our [README](../README.md) to set up Tangle locally.
+
+Always verify your change locally before submitting a PR.
+
+## Tips for high-quality issues
+
+### Use templates
+
+Select an issue template (Bug report, Feature request, etc.). If no template applies, select "Blank issue" and provide as much detail as possible.
+
+### Clear is kind
+
+Clarity saves time, accelerates solutions, and builds community.
+
+<details>
+<summary>Click to see examples of "clear vs. unclear" reporting</summary>
+
+#### Vague vs. Specific Titles
+
+- **Unclear 👎:** "Pipeline doesn't work"
+- **Clear 👍:** "[Bug] Pipeline execution fails with 'Connection refused' error when using custom container images"
+
+#### Vague vs. Specific Steps
+
+- **Unclear 👎:** "I tried to run my pipeline and it didn't work. Please fix."
+- **Clear 👍:** "**Steps to Reproduce:** 1. Create a pipeline with a custom Python component. 2. Configure the component's input argument with a multiline string. 3. Click 'Run Pipeline'. **Expected:** The pipeline submits and runs successfully. **Actual:** The component fails to parse the input and shows a `TypeError` in the execution logs."
+
+#### Vague vs. Specific Feature Requests
+
+- **Unclear 👎:** "You need to add better error messages."
+- **Clear 👍:** "**User Story:** As a data scientist, I need to see detailed validation errors when my component configuration is invalid, so I can fix issues before running a pipeline. Currently, the editor silently ignores invalid YAML in component specs."
+
+</details>
+
+## Tips for high-quality pull requests
+
+### Start as a draft
+
+Start your pull request as a draft. When the automated checks have passed, and you are confident in the changes as well as the description, then remove the draft to enter review.
+
+### Small and focused
+
+A small, focused pull request is easier to understand and review, which means it can be approved and merged much more quickly.
+
+### Write or update tests
+
+Tests prove that your change works correctly and prevent it from breaking in the future.
+
+- Use `npm run test` to run unit tests (**Vitest**).
+- Use `npm run test:e2e:ci` to run end-to-end tests (**Playwright**).
+
+### Match existing code style
```

Comment:
Instructions for formatting (if needed).

Reply:
Added commands